app/controllers/api/host_aggregates_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  class HostAggregatesController < BaseController
+    include Subcollections::Tags
+
+    def create_resource(_type, _id, data = {})
+      ext_management_system = resource_search(data['ems_id'], :providers, collection_class(:providers))
+      data.delete('ems_id')
+
+      task_id = ext_management_system.create_host_aggregate_queue(session[:userid], data)
+      action_result(true, "Creating Host Aggregate #{data['name']} for Provider: #{ext_management_system.name}", :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+
+    def edit_resource(type, id, data = {})
+      raise BadRequestError, "Must specify an id for editing a #{type} resource" unless id
+
+      host_aggregate = resource_search(id, type, collection_class(:host_aggregates))
+      task_id = host_aggregate.update_aggregate_queue(current_user.userid, data)
```

Comment:
@Fryguy this looks similar to the cloud_tenant API that you had a question about (if PATCH should update the provider resource rather than the db record, or should this be an action exposed via POST)

Reply:
Resolving based on the discussion and we will keep this as is.